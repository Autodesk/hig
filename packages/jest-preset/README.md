# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
