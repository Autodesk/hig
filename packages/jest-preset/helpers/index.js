/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
