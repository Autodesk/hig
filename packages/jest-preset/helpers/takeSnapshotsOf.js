const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
