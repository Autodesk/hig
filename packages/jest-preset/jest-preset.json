{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
