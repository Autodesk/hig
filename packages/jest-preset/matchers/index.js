require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
