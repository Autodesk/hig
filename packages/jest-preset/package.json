{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
