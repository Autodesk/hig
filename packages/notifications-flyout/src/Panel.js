import React from "react";
import PropTypes from "prop-types";

import PanelBehavior from "./behaviors/PanelBehavior";
import PanelPresenter from "./presenters/PanelPresenter";

export default function Panel(props) {
  const {
    children,
    heading,
    innerRef,
    loading,
    onScroll,
    transitionStatus,
    stylesheet
  } = props;

  return (
    <PanelBehavior loading={loading} transitionStatus={transitionStatus}>
      {({ listMaxHeight, loadingTransitionState, refListWrapper }) => (
        <PanelPresenter
          heading={heading}
          innerRef={innerRef}
          listMaxHeight={listMaxHeight}
          loadingTransitionState={loadingTransitionState}
          onScroll={onScroll}
          refListWrapper={refListWrapper}
          stylesheet={stylesheet}
        >
          {children}
        </PanelPresenter>
      )}
    </PanelBehavior>
  );
}

Panel.propTypes = {
  children: PropTypes.node,
  heading: PropTypes.string,
  innerRef: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  onScroll: PropTypes.func,
  stylesheet: PropTypes.func,
  transitionStatus: PropTypes.string
};
