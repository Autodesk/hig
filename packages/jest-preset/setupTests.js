require("jest-enzyme");
require("jest-extended");
require("./matchers");
