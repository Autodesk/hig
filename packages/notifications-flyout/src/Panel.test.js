import { takeSnapshotsOf } from "@weave-design/jest-preset/helpers";

import Panel from "./Panel";

describe("notifications-flyout/Panel", () => {
  function handleRef() {}

  takeSnapshotsOf(Panel, [
    {
      desc: "renders without props",
      props: {
        innerRef: handleRef,
      },
    },
    {
      desc: "renders with all props",
      props: {
        children: "foobar",
        heading: "hello",
        innerRef: handleRef,
        loading: true,
        onScroll: function handleScroll() {},
      },
    },
  ]);
});
