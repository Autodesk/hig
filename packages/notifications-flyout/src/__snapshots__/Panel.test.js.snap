// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications-flyout/Panel renders with all props 1`] = `
.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  border: 1px solid transparent;
  box-shadow: 0 0 16px rgba(0,0,0,0.25);
}

.emotion-0 {
  position: relative;
}

<div
  className="emotion-1 hig__flyout-v1__panel-container"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-0 hig__flyout-v1__panel-container__inner"
  >
    <header
      className="hig__notifications-flyout__panel__title"
    >
      hello
    </header>
    <section
      className="hig__notifications-flyout__panel__container"
      style={
        Object {
          "maxHeight": "",
        }
      }
    >
      <div
        onScroll={[Function]}
        role="list"
      >
        foobar
      </div>
    </section>
    <footer
      className="hig__notifications-flyout__panel__loading hig__notifications-flyout__panel__loading--open"
    >
      <div
        data-props="{
  \\"size\\": \\"s\\"
}"
      />
    </footer>
  </div>
</div>
`;

exports[`notifications-flyout/Panel renders without props 1`] = `
.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  border: 1px solid transparent;
  box-shadow: 0 0 16px rgba(0,0,0,0.25);
}

.emotion-0 {
  position: relative;
}

<div
  className="emotion-1 hig__flyout-v1__panel-container"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-0 hig__flyout-v1__panel-container__inner"
  >
    <header
      className="hig__notifications-flyout__panel__title"
    >
      Notifications
    </header>
    <section
      className="hig__notifications-flyout__panel__container"
      style={
        Object {
          "maxHeight": "",
        }
      }
    >
      <div
        onScroll={undefined}
        role="list"
      />
    </section>
    <footer
      className="hig__notifications-flyout__panel__loading"
    >
      <div
        data-props="{
  \\"size\\": \\"s\\"
}"
      />
    </footer>
  </div>
</div>
`;
